activemodel/lib/active_model/nested_attributes.rb

```diff
@@ -0,0 +1,572 @@
+    # hash, you have to enable it first using the <tt>:allow_destroy</tt>
+    # option. This will allow you to also use the <tt>_destroy</tt> key to
+    # destroy existing models:
+    #
+    #   class Member
+    #     include ActiveModel::Model
+    #     include ActiveModel::NestedAttributes
+    #
+    #     attr_accessor :posts
+    #     accepts_nested_attributes_for :posts, class_name: Post, allow_destroy: true
+    #   end
+    #
+    #   params = { member: {
+    #     posts_attributes: [{ _destroy: '1' }]
+    #   }}
+    #
+    #   member.attributes = params[:member]
+    #   member.posts.length # => 0
+    #
+    # If you add the <tt>_destroy</tt> key to the attributes hash and the
+    # assigned instances respond to <tt>mark_for_destruction?</tt>, assignment
+    # will call that method on the associated models:
+    #
+    #   params = { member: {
+    #     posts_attributes: [{ _destroy: '1' }]
+    #   }}
+    #   member.posts.length # => 1
+    #   member.posts.first.marked_for_destruction? # => true
+    #
+    # Note that the models will _not_ be unassigned automatically. It is the
+    # parent's responsibility to remove models marked for destruction.
+    #
+    # Nested attributes for an associated collection can also be passed in
+    # the form of a hash of hashes instead of an array of hashes:
+    #
+    #   Member.new(
+    #     name: 'joe',
+    #     posts_attributes: {
+    #       first:  { title: 'Foo' },
+    #       second: { title: 'Bar' }
+    #     }
+    #   )
+    #
+    # has the same effect as
+    #
+    #   Member.new(
+    #     name: 'joe',
+    #     posts_attributes: [
+    #       { title: 'Foo' },
+    #       { title: 'Bar' }
+    #     ]
+    #   )
+    #
+    # The keys of the hash which is the value for +:posts_attributes+ are
+    # ignored in this case.
+    #
+    # Passing attributes for an associated collection in the form of a hash
+    # of hashes can be used with hashes generated from HTTP/HTML parameters,
+    # where there may be no natural way to submit an array of hashes.
+    #
+    # === Integration with ActiveModel::Attributes
+    #
+    # When attributes are declared with the +.attribute+ class method provided
+    # by the Attributes module, support
+    # for nested attributes will construct instances using available
+    # Type information. For example, the +accepts_nested_attributes_for+
+    # declarations in the following code sample will use the Type that corresponds
+    # to +:user+ when assigning to +author_attributes+ and the Type that
+    # correpsonds to +:tags+ when assigning to +tags_attributes+:
+    #
+    #   class Article
+    #     include ActiveModel::Model
+    #     include ActiveModel::Attributes
+    #     include ActiveModel::NestedAttributes
+    #
+    #     attribute :author, :user
+    #     attribute :tags, :tag
+    #
+    #     accepts_nested_attributes_for :author
+    #     accepts_nested_attributes_for :tags
+    #   end
```

Comment:
@rafaelfranca in reference to https://github.com/rails/rails/pull/49637#issuecomment-1769034124:

Does this style of definition match what you have in mind?

The implementation builds on top of https://github.com/rails/rails/pull/49910 (authored by @jonathanhefner and @p8) to first try utilizing defined Attributes before making any kind of inference to classes to build from Hashes:

* https://github.com/rails/rails/pull/49637/commits/9943b8a4eb971b27f1139ee692ad4a2242d8a0e7#diff-631152b72279c286a5d659f39fbbe530e801049c8f82b5192087cff6aeffa837R442-R443

* https://github.com/rails/rails/pull/49637/commits/9943b8a4eb971b27f1139ee692ad4a2242d8a0e7#diff-631152b72279c286a5d659f39fbbe530e801049c8f82b5192087cff6aeffa837R481-R488